x-pack/test/observability_onboarding_api_integration/tests/flow/progress/progress.spec.ts

```diff
@@ -133,106 +138,224 @@ export default function ApiTest({ getService }: FtrProviderContext) {
             });
 
             describe('when logs have been ingested', () => {
-              describe('and onboarding type is logFiles', () => {
-                before(async () => {
-                  await es.indices.createDataStream({
-                    name: `logs-${datasetName}-${namespace}`,
+              describe('with a different agentId', () => {
+                describe('and onboarding type is logFiles', () => {
+                  before(async () => {
+                    await es.indices.createDataStream({
```

Comment:
Can we please move this `datastream` creation and `log doc` creation to a parameterized helper method as i am seeing we are doing 4 times in this file.

With this once we have synthtrace, it will be easier to replace the logic. 

Also move the clean index logic to a helper which takes the ds as paramter

Reply:
I prefer to wait on this one for synthtrace replacement, wdyt?
My idea would be to replace all of this `log` [creation|handling|removal] with it 😇